### DISCUSSION TITLE
Filter DropDown Values DataTables

### DISCUSSION THREAD

**Discussion by GaurangDhorda** [l-lin/angular-datatables#1538] [Q&A]
[StackBllitz Demo](https://stackblitz.com/edit/drop-down-population-joql41?file=app%2Fapp.component.html)

in this demo I am filter table data using table footer.. and table body has one column called dropdowns. I want to filter out all selected values in column when filter out by. but it will not work, if you disable app.component.html line no 39 select then every thing working fine. How can filter searching inside dropdown values ? So why that.search() not working for select values.

**Comment by shanmukhateja** [comment]
So, if I understand correctly, you're trying to trigger table search from `<select>` on the table's footer but I do not understand why you're including another `<select>` element inside `<td>` node? Please explain.

**Reply by GaurangDhorda** [reply]
If you see running output, first column of table is `checkbox`, when you checked this checkbox dropdown from table `<td>` enables, when we select any value lets say `Superman` inside table, and doing same for many row, then in footer we filter by selecting `superman` then it should filter out all table dropdown value selected with `Superman`...

**Reply by GaurangDhorda** [reply]
@shanmukhateja Any ways How can I achieve this kind of functionality ? Thank You.

**Reply by shanmukhateja** [reply]
Tried to get it work but it doesn't work :(

**Reply by GaurangDhorda** [reply]
@shanmukhateja any other way you can suggests to do same thing, or angular-table is not supporting this kind of ways. Thank You So much for your kind attention to this problem statement .

**Reply by shanmukhateja** [reply]
I don't know why it refuses to work when using `<option>`. @l-lin if you're free, could you take a look?

**Reply by l-lin** [reply]
> So why that.search() not working for select values.

That's because the content of your column is not just values, but the whole HTML select content.

This [discussion in DataTables forum](https://datatables.net/forums/discussion/49793/column-search-with-an-html-tag) may help you.

**Reply by GaurangDhorda** [reply]
@l-lin  Thank You for your kind attention.  Then How can we search inside whole HTML select content ? Can you provide any example code to do it ? How can we do the same in angular ? inside ngAfterViewInit()

**Reply by l-lin** [reply]
One way is to use the [search API](https://datatables.net/reference/api/column().search()) with regex to find out the selected value in your list.

**Reply by GaurangDhorda** [reply]
@l-lin https://stackblitz.com/edit/drop-down-population-q5hqit?file=app%2Fapp.component.ts as shown in this example if you check app.component.ts line number of 112 then, regex not detecting when we select value from footer @shanmukhateja and if we type value in 3rd input-box inside footer of table than it will search for dropdown column too, if you type superman, then it will also filter out select column of  table too. But same is not working when I select value inside footer select
